src/control/server/mgmt_pool.go

```diff
@@ -482,6 +482,14 @@ func (svc *mgmtSvc) poolCreate(parent context.Context, req *mgmtpb.PoolCreateReq
 		return nil, errors.Wrap(err, "unmarshal PoolCreate response")
 	}
 
+	// Zero mem_file_bytes in non-MD-on-SSD mode.
+	if !svc.harness.Instances()[0].GetStorage().BdevRoleMetaConfigured() {
+		resp.MemFileBytes = 0
```

Comment:
Backing up a bit, though, why is this being fiddled with in the control plane? Shouldn't the engine set this correctly based on its configuration? Having logic split across control/data planes should be avoided when possible, as it's harder to understand and maintain.

Reply:
I had to choose between "fiddling with it" in the engine or the control-plane-server in order to be able to detect display type on the control-plane-client side (dmg). I didn't want to change within the engine because it feels better to report a real value for memory file size regardless of MD-on-SSD mode (the general trend within dRPC responses is that the presence of most of the values don't vary much for PMem/MD-on-SSD mode). Memory-file-bytes will equal SCM size in PMem, reporting this engine side as zero for PMem size seems counterintuitive and for some reason toggling on the control-plane server side seems more reasonable to me.

Reply:
Looking at the pool query flow, in order to do this adjustment within srv_drpc.c we would have to pull in bio.h and that might not be desirable. This being said is it not a better compromise to leave the adjustments where they are?